src/dotnet/commands/dotnet-migrate/MigrateCommand.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.ProjectJsonMigration;
+
+namespace Microsoft.DotNet.Tools.Migrate
+{
+    public partial class MigrateCommand
+    {
+        private string _templateFile; 
+        private string _outputDirectory;
+        private string _projectJson;
+        private string _sdkVersion;
+
+        private TemporaryDotnetNewTemplateProject _temporaryDotnetNewProject;
+
+        public MigrateCommand(string templateFile, string outputDirectory, string projectJson, string sdkVersion)
+        {
+            _templateFile = templateFile;
+            _outputDirectory = outputDirectory;
+            _projectJson = projectJson;
+            _sdkVersion = sdkVersion;
+
+            _temporaryDotnetNewProject = new TemporaryDotnetNewTemplateProject();
+        }
+
+        public int Start()
```

Comment:
Or "Execute".

---
In reply to: [75899422](https://github.com/dotnet/cli/pull/4074#discussion_r75899422) [](ancestors = 75899422)